docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+##### Paramaterization
+
+Named functions can be passed to other functions as a parameter as follows\:
+```
+func void saySomething = () =>{ print("something"); };
+func void doSomething = (func f) => { f(); };
+doSomething(saySomething);
+```
+
+##### Caveats
+
+* Fire does not support function overloading
+* Fire does not support genericity in functions
+
+#### 5.1.4. `array`
+
+The `array` type is a dynamic collection of elements. Inspired by AWK's associative arrays, an `array` collection maps keys of one type to values of one type. Keys and values do not have be the same type, but all keys must share the same type and all values must share the same type.
+
+The structure of `array` variable declarations is as follows:
+
+`array arr[<key_type>, <value_type>];`
+
+Example:
+
+`array arr[int, string];` 
+
+The assignment of variables has the following structure:
+
+`arr[<key_value>] = <element>;` 
+
+Example:
+
+ `arr[17] = "age17";`
+ 
+ Finally, a programmer can retrieve a value associated with a key with the below syntax:
+ 
+ `int element = arr[<key_value>];`
+ 
+ Example:
+ 
+ `int age = arr["age"];`
+ 
+#### 5.1.5. `regx`
+
+Regular expressions are supported in FIRE. Via the `regx` type, which assigns an object to a regular expression. That object can then be passed as a parameter to functions that utilize regular expressions to a pattern match or extract data.
+
+The structure of a `regx` declaration is as follows:
+
+`regx myPattern = r'<pattern>'`
+
+Example:
+
+```
+regx myPattern = r'[a-z]';
+myFunction(someString, myPattern);
+```
+
+The syntax for the regex patterns are as follows:
+
+* `\` escapes any of the operators for the literal character
+* `^` matches only the beginning of the string
+* `$` matches only the end of the string
+* `.` matches any single character
+* `[ ... ]` defines a character list, where the character list can also be character range. This matches any string containing these characters
+* `[^ ... ]` defines a character list, but negates them. This matches any string *not* containing these characters
+* `|` matches either expression `e1` or `e2`
+* `( ... )` groups expressions together where `...` is some regular expression
+* `*` matches the preceding character 0 or many times
+* `+` matches the preceding character at least once
+
+#### 5.1.6. `bool`
+
+Boolean objects contain a value of either `true` or `false`. They can be declared on their own, and are used in conditional statements.
+
+The structure of a boolean declaration:
+
+```
+bool switch = true; // or false
+```
+
+Example: 
+
+```
+int x = 0;
+bool switch = true;
+
+// infinite loop
+while(switch) {
+	x = x + 1;
+}
+```
+
+## 6: Statements 
+
+#### 6.1 Print Statement
+
+The print statement prints a literal value, or the value returned by an expression.
+
+`print("i will be printed to stdout");`
+
```

Comment:
can you print integers? can you print `array`s? `regx`s? `array`s?

you need to give more explicit typing constraints

Reply:
need to address if we need to be more explicit or not